app/src/main/java/com/celzero/bravedns/ui/HomeScreenFragment.kt

```diff
@@ -299,6 +302,64 @@ class HomeScreenFragment : Fragment(R.layout.fragment_home_screen) {
         }.start()
     }
 
+    private fun showStopDialogAlwaysOn() {
+        val builder = AlertDialog.Builder(requireContext())
+        var isLockDownEnabled = false
+        if (VERSION.SDK_INT >= VERSION_CODES.Q) {
+            val vpnService = VpnController.getInstance().getBraveVpnService()
+            if (vpnService != null) {
+                isLockDownEnabled = vpnService.isLockdownEnabled
+            }
+        }
+        //set title for alert dialog
+        builder.setTitle(R.string.always_on_dialog_stop_heading)
+        if (isLockDownEnabled) {
+            //set message for alert dialog
+            if (VERSION.SDK_INT >= VERSION_CODES.N) {
+                builder.setMessage(Html.fromHtml(getString(R.string.always_on_dialog_lockdown_stop_message), HtmlCompat.FROM_HTML_MODE_COMPACT))
+            } else {
+                builder.setMessage(HtmlCompat.fromHtml(getString(R.string.always_on_dialog_lockdown_stop_message), HtmlCompat.FROM_HTML_MODE_COMPACT))
+            }
+        } else {
+            //set message for alert dialog
+            builder.setMessage(R.string.always_on_dialog_stop_message)
+        }
+
+        builder.setCancelable(true)
+        //performing positive action
+        builder.setPositiveButton(R.string.always_on_dialog_positive) { _, _ ->
+            stopVpnService()
+        }
+
+        builder.setNegativeButton(R.string.always_on_dialog_negative) { _, _ ->
+        }
+
+        builder.setNeutralButton(R.string.always_on_dialog_neutral) { _, _ ->
+            openVPNProfile()
+        }
+
+        // Create the AlertDialog
+        val alertDialog: AlertDialog = builder.create()
+        // Set other dialog properties
+        alertDialog.setCancelable(false)
+        alertDialog.show()
+    }
+
+    private fun openVPNProfile() {
+        try {
+            val intent = if (VERSION.SDK_INT >= VERSION_CODES.N) {
+                Intent(ACTION_VPN_SETTINGS)
+            } else {
+                Intent("android.net.vpn.SETTINGS")
```

Comment:
this should open RethinkDNS section, but it opens the generic VPN profile page.

Reply:
Android doesn't provide API to add the package-name in Intent's data to open the specific RethinkDNS section. So it opens the generic VPN profile page.